### PR TITLE
Moving GC Followup

### PR DESCRIPTION
**Pull request by vyzo**
Follow up from #6854 

This improves code readability with better variable names.
It also fixes a (minor) bug, whereby we always create absolute symlinks even though the link might be in the same directory

### CODE REVIEW

File: blockstore/badger/blockstore.go [hunk 1]
```diff
@@ -296,68 +297,96 @@ func (b *Blockstore) movingGC() error {
...
 
-	b.deleteDB(oldpath)
+	// the delete follows symlinks
+	b.deleteDB(backupPath)
 
 	log.Info("moving blockstore done")
 	return nil
 }
 
+// symlink creates a symlink from path to linkPath; the link is relative if the two are
```
Comment (Kubuxu) [inline blockstore/badger/blockstore.go:370]:
```suggestion
// symlink creates a symlink from path to linkTo; the link is relative if the two are
```

File: blockstore/badger/blockstore.go [hunk 2]
```diff
@@ -296,68 +297,96 @@ func (b *Blockstore) movingGC() error {
...
+// in the same directory
+func (b *Blockstore) symlink(path, linkTo string) error {
+	resolvedPathDir, err := filepath.EvalSymlinks(filepath.Dir(path))
```
Comment (Kubuxu) [inline blockstore/badger/blockstore.go:373]: shouldn't you first eval symlinks and then take the dir?

Comment (vyzo) [reply blockstore/badger/blockstore.go:373]: No, linkTo doesn't exist yet.

Comment (vyzo) [reply blockstore/badger/blockstore.go:373]: We could do it for path however, just kept it like this for symmetry.

Comment (vyzo) [reply blockstore/badger/blockstore.go:373]: But we know that it is not a symlink itself, as we created, so moot.

File: blockstore/badger/blockstore.go [hunk 3]
```diff
@@ -296,68 +297,96 @@ func (b *Blockstore) movingGC() error {
...
+	// the delete follows symlinks
+	b.deleteDB(backupPath)
 
 	log.Info("moving blockstore done")
 	return nil
 }
 
+// symlink creates a symlink from path to linkTo; the link is relative if the two are
+// in the same directory
+func (b *Blockstore) symlink(path, linkTo string) error {
```
Comment (Stebalien) [inline blockstore/badger/blockstore.go:372]: make this a free function?

**Review by Stebalien — APPROVED** [review]

File: blockstore/badger/blockstore.go (hunk 3 above)
Comment (vyzo) [reply blockstore/badger/blockstore.go:372]: sure!

Comment (vyzo) [reply blockstore/badger/blockstore.go]: done.

**Comment by jennijuju** [comment]
This should be backported to release/v1.11.1

**Pull request merged**